core/modules/node/node.module

```diff
@@ -2665,6 +2665,7 @@ function node_form_system_themes_admin_form_alter(&$form, &$form_state, $form_id
  */
 function node_form_system_themes_admin_form_submit($form, &$form_state) {
   config_set('system.core', 'node_admin_theme', $form_state['values']['node_admin_theme']);
+  cache('layout_path')->flush();
```

Comment:
Just to be comprehensive, maybe we should use Layout module's dedicated cache clear function? Since themes and layouts are so tightly coupled, it might prevent other issues in the future.
```suggestion
  // The default and admin layout are tied directly to this setting, so clear
  // the Layout cache.
  layout_reset_caches()
```

Reply:
Hm. This makes this switching that single setting "more expensive".

But you're right, they're tightly coupled, so better flush all, including all layout config. Switching this setting doesn't happen very often, I guess.

Reply:
Oops, where's the semicolon. :speak_no_evil: